agent/reconciler/host_reconciler.go

```diff
@@ -117,10 +121,20 @@ func (r *HostReconciler) reconcileNormal(ctx context.Context, byoHost *infrastru
 		if r.SkipK8sInstallation {
 			logger.Info("Skipping installation of k8s components")
 		} else if r.UseInstallerController {
-			if byoHost.Spec.InstallationSecret == nil {
-				logger.Info("K8sInstallationSecret not ready")
-				conditions.MarkFalse(byoHost, infrastructurev1beta1.K8sNodeBootstrapSucceeded, infrastructurev1beta1.K8sInstallationSecretUnavailableReason, clusterv1.ConditionSeverityInfo, "")
-				return ctrl.Result{}, nil
+			if !conditions.IsTrue(byoHost, infrastructurev1beta1.K8sComponentsInstallationSucceeded) {
```

Comment:
Looks like a similar condition is needed on line#140 as well. But given that we will deprecate it pretty soon, we might be okay at this point.
cc @mayur-tolexo

Reply:
That means no changes required. right?